featuretools/primitives/standard/transform_primitive.py

```diff
@@ -906,3 +907,138 @@ def lag(time_index, numeric):
             x = pd.Series(numeric.values, index=time_index.values)
             return x.shift(periods=self.periods, fill_value=self.fill_value).values
         return lag
+
+
+class IsInGeoBox(TransformPrimitive):
+    """Determines if coordinates are inside a box defined by two
+    corner coordinate points.
+
+    Description:
+        Coordinate values should be specified as (latitude, longitude)
+        tuples. This primitive is unable to handle coordinates and boxes
+        at the poles, and near +/- 180 degrees latitude.
+
+    Args:
+        point1 (tuple(float, float)): The coordinates
+            of the first corner of the box. Defaults to (0, 0).
+        point2 (tuple(float, float)): The coordinates
+            of the diagonal corner of the box. Defaults to (0, 0).
+
+    Example:
+        >>> is_in_geobox = IsInGeoBox((40.7128, -74.0060), (42.2436, -71.1677))
+        >>> is_in_geobox([(41.034, -72.254), (39.125, -87.345)]).tolist()
+        [True, False]
+    """
+    name = "is_in_geobox"
+    input_types = [ColumnSchema(logical_type=LatLong)]
+    return_type = ColumnSchema(logical_type=BooleanNullable)
+
+    def __init__(self, point1=(0, 0), point2=(0, 0)):
+        self.point1 = point1
+        self.point2 = point2
+        self.lats = np.sort(np.array([point1[0], point2[0]]))
+        self.lons = np.sort(np.array([point1[1], point2[1]]))
+
+    def get_function(self):
+        def geobox(latlongs):
+            if latlongs.hasnans:
+                latlongs = np.where(latlongs.isnull(), pd.Series([(np.nan, np.nan)] * len(latlongs)), latlongs)
+            transposed = np.transpose([list(latlon) for latlon in latlongs])
+            lats = (self.lats[0] <= transposed[0]) & \
+                   (self.lats[1] >= transposed[0])
+            longs = (self.lons[0] <= transposed[1]) & \
+                    (self.lons[1] >= transposed[1])
+            return lats & longs
+        return geobox
+
+
+class GeoMidpoint(TransformPrimitive):
+    """Determines the geographic center of two coordinates.
+
+    Examples:
+        >>> geomidpoint = GeoMidpoint()
+        >>> geomidpoint([(42.4, -71.1)], [(40.0, -122.4)])
+        [(41.2, -96.75)]
+    """
+    name = "geomidpoint"
+    input_types = [ColumnSchema(logical_type=LatLong), ColumnSchema(logical_type=LatLong)]
+    return_type = ColumnSchema(logical_type=LatLong)
+    commutative = True
+
+    def get_function(self):
+        def geomidpoint_func(latlong_1, latlong_2):
+            lat_1s = np.array([x[0] if isinstance(x, tuple) else np.nan for x in latlong_1])
+            lon_1s = np.array([x[1] if isinstance(x, tuple) else np.nan for x in latlong_1])
```

Comment:
This taking a latlong series and deconstructing it into lats and longs now happens 6 different times across the latlong primitives; it's a lot of repeated code. Can we pull it out into a helper util `lat_1s, lon_1s = deconstrct_latlongs(latlong_1)` or something?

Reply:
Done
https://github.com/alteryx/featuretools/pull/1814/commits/d83d2c6d2ac6ecec01932825866c7abab7da5af6